src/protagonist/DLCS.Repository/Assets/BatchRepository.cs

```diff
@@ -44,4 +45,23 @@ public async Task<Batch> CreateBatch(int customerId, IReadOnlyList<Asset> assets
 
         return batch;
     }
+    
+    /// <inheritdoc />
+    public async Task<BatchAsset> CreateBatchAsset(Asset asset, CancellationToken cancellationToken)
+    {
+        asset.Batch.ThrowIfNull("Saving a batch asset requires a batch");

```

Comment:
Argument for `ThrowIfNull` isn't an error message, it's the parameter name for [`ArgumentNullException`](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.-ctor?view=net-8.0#system-argumentnullexception-ctor(system-string))